src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/messaging/PrisonerReceivedEventTest.kt

```diff
@@ -0,0 +1,142 @@
+      .wasStatus(InductionScheduleStatusResponse.SCHEDULED)
+
+    // test induction schedule history is created
+    val inductionScheduleHistories = getInductionScheduleHistory("A6099EA")
+    assertThat(inductionScheduleHistories.inductionSchedules).size().isEqualTo(1)
+    assertThat(inductionScheduleHistories.inductionSchedules[0])
+      .wasCreatedAfter(earliestDateTime)
+      .wasUpdatedAfter(earliestDateTime)
+      .wasCreatedBy("system")
+      .wasCreatedByDisplayName("system")
+      .wasUpdatedBy("system")
+      .wasUpdatedByDisplayName("system")
+      .wasScheduleCalculationRule(InductionScheduleCalculationRuleResponse.NEW_PRISON_ADMISSION)
+      .wasStatus(InductionScheduleStatusResponse.SCHEDULED)
+      .wasVersion(1)
+
+    // test that outbound event is also created:
+    val inductionScheduleEvent = inductionScheduleEventQueue.receiveEvent(QueueType.INDUCTION)
+    assertThat(inductionScheduleEvent.personReference.identifiers[0].value).isEqualTo("A6099EA")
+    assertThat(inductionScheduleEvent.detailUrl).isEqualTo("http://localhost:8080/inductions/A6099EA/induction-schedule")
+  }
+
+  @Test
+  fun `should update the Review Schedule and send outbound message given 'prisoner received' event for prisoner that already has an Induction and initial Review Schedule`() {
+    // Given
+    // an induction schedule, induction and action plan are created. This will have created the initial Review Schedule
+    val prisonNumber = randomValidPrisonNumber()
+    createInductionSchedule(
+      prisonNumber = prisonNumber,
+      status = InductionScheduleStatus.COMPLETE,
+    )
+    createInduction(prisonNumber, aValidCreateInductionRequestForPrisonerNotLookingToWork())
+    createActionPlan(prisonNumber)
+
+    // The above calls set the data up but they will also generate events so clear these out before starting the test.
+    // Before clearing the queues though we need to wait until the first "plp.review-schedule.updated" event on the REVIEW queue is received.
+    await untilCallTo {
+      reviewScheduleEventQueue.countAllMessagesOnQueue()
+    } matches { it != null && it > 0 }
+    clearQueues()
+
+    val expectedPrisoner = aValidPrisoner(prisonerNumber = prisonNumber)
+    createPrisonerAPIStub(prisonNumber, expectedPrisoner)
+
+    val earliestDateTime = OffsetDateTime.now()
+
+    val sqsMessage = prisonerReceivedSqsMessage(prisonNumber)
+
+    // When
+    sendDomainEvent(sqsMessage)
+
+    // Then
+    // wait until the queue is drained / message is processed
+    await untilCallTo {
+      domainEventQueueClient.countMessagesOnQueue(domainEventQueue.queueUrl).get()
+    } matches { it == 0 }
+
+    val actionPlanReviews = getActionPlanReviews(prisonNumber)
+    assertThat(actionPlanReviews)
+      .latestReviewSchedule {
+        it.hasStatus(ReviewScheduleStatus.SCHEDULED)
+          // TODO fix the implementation in PefCiagKpiService.processPrisonerAdmission to make this set the correct calculation rule
```

Comment:
This TODO will also need addressing, but was out of scope for what I was trying to fix here.
But at least the refactoring of this test identified that we need to do it 👍